pandas/core/arrays/string_arrow.py

```diff
@@ -697,9 +697,9 @@ def _reduce(
         self, name: str, *, skipna: bool = True, keepdims: bool = False, **kwargs
     ):
         if name in ["any", "all"]:
-            if not skipna and name == "all":
-                nas = pc.invert(pc.is_null(self._pa_array))
-                arr = pc.and_kleene(nas, pc.not_equal(self._pa_array, ""))
+            if not skipna:
+                nas = pc.is_null(self._pa_array)
+                arr = pc.or_kleene(nas, pc.not_equal(self._pa_array, ""))
```

Comment:
Before, for converting the string array to a boolean array, True values were considered as _"not `""`, and not null"_, and this diff changed that to _"not `""`, or null"_